spec/unit/config_files_inspector_spec.rb

```diff
@@ -96,231 +118,346 @@
-        expect(system).to receive(:retrieve_files).with(
-          config_paths,
-          cfdir
-        )
+
+      it "returns a unique list of packages with config files" do
+        expect(system).to receive(:run_command).with(
+          "rpm", "-qa", "--configfiles", "--queryformat",
+          "%{NAME}-%{VERSION}\n", {:stdout => :capture}
+        ).and_return(rpm_qa_output_test3)
+
+        package_list = subject.packages_with_config_files
+
+        expect(package_list).to match_array expected_package_list
       end
-    end
 
-    it "returns data about modified config files when requirements are fulfilled" do
-      system = double
-      md5_os = ConfigFile.new(
-        name:            "",
-        package_name:    "apache2",
-        package_version: "2.4.6",
-        status:          "changed",
-        changes:         ["size", "md5", "time"],
-        user:            "root",
-        group:           "root",
-        mode:            "644"
-      )
-
-      apache_1 = md5_os.dup
-      apache_1.name = "/etc/apache2/default-server.conf"
-
-      apache_2 = md5_os.dup
-      apache_2.name = "/etc/apache2/listen.conf"
-      apache_2.changes = ["md5", "time"]
-      apache_2.mode = "6644"
-
-      apache_3 = md5_os.dup
-      apache_3.name = "/etc/sysconfig/SuSEfirewall2.d/services/apache2"
-
-      apache_4 = ConfigFile.new(
-        name:            "/usr/share/info/dir",
-        package_name:    "apache2",
-        package_version: "2.4.6",
-        status:          "changed",
-        changes:         ["deleted"]
-      )
-      apache_5 = ConfigFile.new(
-        name:            "/usr/share/man/man1/time.1.gz",
-        package_name:    "apache2",
-        package_version: "2.4.6",
-        status:          "changed",
-        changes:         ["replaced"],
-        user:            "root",
-        group:           "root",
-        mode:            "644"
-      )
-
-      iscsi_1 = md5_os.dup
-      iscsi_1.changes = ["size", "mode", "md5", "user", "group", "time"]
-      iscsi_1.name = "/etc/iscsi/iscsid.conf"
-      iscsi_1.package_name = "open-iscsi"
-      iscsi_1.package_version = "2.0.873"
-      iscsi_1.user = "nobody"
-      iscsi_1.group = "nobody"
-      iscsi_1.mode = "4700"
-
-      expected_data = ConfigFilesScope.new(
-        extracted: false,
-        files: ConfigFileList.new([apache_1, apache_2, iscsi_1, apache_3, apache_4, apache_5])
-      )
-
-      expect_inspect_configfiles(system, false)
-
-      inspector = ConfigFilesInspector.new(system, description)
-      inspector.inspect(filter)
-
-      expect(description["config_files"]).to eq(expected_data)
-      expect(inspector.summary).to include("6 changed configuration files")
+      it "returns a empty list when no packages contain config files" do
+        expect(system).to receive(:run_command).with(
+          "rpm", "-qa", "--configfiles", "--queryformat",
+          "%{NAME}-%{VERSION}\n", {:stdout => :capture}
```

Comment:
Redundant curly braces around a hash parameter.<br>Use the new Ruby 1.9 hash syntax.<br>Space inside { missing.<br>Space inside } missing.